dash/examples/ex.11.simple-stencil/main.cpp

```diff
@@ -0,0 +1,221 @@
+  bool is_left   =(local_beg_gidx[0] == 0) ? true : false;
+  bool is_right  =(local_end_gidx[0] == (gext_x-1)) ? true : false;
+
+  // inner-top
+  if(!is_top){
+    for( auto x=begin_idx_x; x<end_idx_x; ++x){
+      nlptr[lext_y*x] =
+        ( 0.40 * olptr[lext_y*x] +
+        0.15 * data_old.at(local_beg_gidx[0] + x,   local_beg_gidx[1]-1) +
+        0.15 * data_old.at(local_beg_gidx[0] + x,   local_beg_gidx[1]+1) +
+        0.15 * data_old.at(local_beg_gidx[0] + x+1, local_beg_gidx[1]) +
+        0.15 * data_old.at(local_beg_gidx[0] + x-1, local_beg_gidx[1]));
+    }
+  }
+  if(!is_bottom){
+    for( auto x=begin_idx_x; x<end_idx_x; ++x){
+      nlptr[lext_y*x + end_idx_y] =
+        ( 0.40 * olptr[lext_y*x + end_idx_y] +
+        0.15 * data_old.at(local_beg_gidx[0] + x,   local_end_gidx[1]-1) +
+        0.15 * data_old.at(local_beg_gidx[0] + x,   local_end_gidx[1]+1) +
+        0.15 * data_old.at(local_beg_gidx[0] + x+1, local_end_gidx[1]) +
+        0.15 * data_old.at(local_beg_gidx[0] + x-1, local_end_gidx[1]));
+    }
+  }
+  if(!is_left){
+    for( auto y=begin_idx_y; y<end_idx_y; ++y){
+      nlptr[y] =
+        ( 0.40 * olptr[y] +
+        0.15 * data_old.at(local_beg_gidx[0]-1, local_beg_gidx[1] + y) +
+        0.15 * data_old.at(local_beg_gidx[0]+1, local_beg_gidx[1] + y) +
+        0.15 * data_old.at(local_beg_gidx[0],   local_beg_gidx[1] + y-1) +
+        0.15 * data_old.at(local_beg_gidx[0],   local_beg_gidx[1] + y+1));
+    }
+  }
+  if(!is_right){
+    for( auto y=begin_idx_y; y<end_idx_y; ++y){
+      nlptr[lext_y * end_idx_x + y] =
+        ( 0.40 * olptr[lext_y * end_idx_x + y] +
+        0.15 * data_old.at(local_end_gidx[0]-1, local_beg_gidx[1] + y) +
+        0.15 * data_old.at(local_end_gidx[0]+1, local_beg_gidx[1] + y) +
+        0.15 * data_old.at(local_end_gidx[0],   local_beg_gidx[1] + y-1) +
+        0.15 * data_old.at(local_end_gidx[0],   local_beg_gidx[1] + y+1));
+    }
+  }
+}
+
+int main(int argc, char* argv[])
+{
+  int sizex = 1000;
+  int sizey = 1000;
+  int niter = 20;
+
+  dash::init(&argc, &argv);
+  
+  // Prepare grid
+  dash::TeamSpec<2> ts;
+  dash::SizeSpec<2> ss(sizex, sizey);
+  dash::DistributionSpec<2> ds(dash::BLOCKED, dash::BLOCKED);
+  ts.balance_extents();
+
+  dash::Pattern<2> pattern(ss, ds, ts);
+
+  Array_t data_old(pattern);
+  Array_t data_new(pattern);
+
+  dash::fill(data_old.begin(), data_old.end(), 255);
+  dash::fill(data_new.begin(), data_new.end(), 255);
+
+  draw_circle(data_old, 0, 0, 40);
+  draw_circle(data_old, 0, 0, 30);
+  draw_circle(data_old, 100, 100, 10);
+  draw_circle(data_old, 100, 100, 20);
+  draw_circle(data_old, 100, 100, 30);
+  draw_circle(data_old, 100, 100, 40);
+  draw_circle(data_old, 100, 100, 50);
+
+  dash::barrier();
+  write_pgm("testimg_input.pgm", data_old);
+  dash::barrier();
+
+  for(int i=0; i<niter; ++i){
+    smooth(data_old, data_new);
+    dash::barrier();
+    std::copy(data_new.lbegin(), data_new.lend(), data_old.lbegin());
```

Comment:
The interface of dash::copy does not support copying a GlobIter to another GlobIter.